secretstores/akeyless/akeyless.go

```diff
@@ -0,0 +1,539 @@
+
+	// separate items by type since only static secrets are supported for bulk get
+	staticItems, dynamicItems, rotatedItems := a.separateItemsByType(listItems)
+	a.logger.Infof("%d items returned (static: %d, dynamic: %d, rotated: %d)", len(listItems), len(staticItems), len(dynamicItems), len(rotatedItems))
+
+	// listItems can get quite large, so we don't need all item details, we can use the item names instead
+	// and free memory
+	listItems = nil
+	staticItemNames := GetItemNames(staticItems)
+	dynamicItemNames := GetItemNames(dynamicItems)
+	rotatedItemNames := GetItemNames(rotatedItems)
+	a.logger.Debugf("static items: %v", staticItemNames)
+	a.logger.Debugf("dynamic items: %v", dynamicItemNames)
+	a.logger.Debugf("rotated items: %v", rotatedItemNames)
+
+	haveStaticItems := len(staticItemNames) > 0
+	haveDynamicItems := len(dynamicItemNames) > 0
+	haveRotatedItems := len(rotatedItemNames) > 0
+
+	secretResultChannels := make(chan secretResultCollection, boolToInt(haveStaticItems)+boolToInt(haveDynamicItems)+boolToInt(haveRotatedItems))
+
+	mutex := sync.Mutex{}
+
+	// get secret values concurrently, each item type in a separate goroutine
+	wg := sync.WaitGroup{}
+	if haveStaticItems {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			if len(staticItemNames) == 1 {
+				staticSecretName := staticItemNames[0]
+				value, err := a.GetSingleSecretValue(ctx, staticSecretName, STATIC_SECRET_RESPONSE)
+				secretResultChannels <- secretResultCollection{name: staticSecretName, value: value, err: err}
+			} else {
+				secretResponse := a.GetBulkStaticSecretValues(ctx, staticItemNames)
+				if len(secretResponse) > 0 {
+					for _, result := range secretResponse {
+						secretResultChannels <- result
+					}
+				}
+			}
+		}()
+	}
+	if haveDynamicItems {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for _, item := range dynamicItemNames {
+				value, err := a.GetSingleSecretValue(ctx, item, DYNAMIC_SECRET_RESPONSE)
+				if err != nil {
+					secretResultChannels <- secretResultCollection{name: item, value: "", err: err}
+				} else {
+					secretResultChannels <- secretResultCollection{name: item, value: value, err: nil}
+				}
+			}
+		}()
+	}
+	if haveRotatedItems {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for _, item := range rotatedItemNames {
+				value, err := a.GetSingleSecretValue(ctx, item, ROTATED_SECRET_RESPONSE)
+				if err != nil {
+					secretResultChannels <- secretResultCollection{name: item, value: "", err: err}
+				} else {
+					secretResultChannels <- secretResultCollection{name: item, value: value, err: nil}
+				}
+			}
+		}()
+	}
+
+	// close the channel when all goroutines are done
+	go func() {
+		wg.Wait()
+		close(secretResultChannels)
+	}()
+
+	// collect results and populate response
+	for result := range secretResultChannels {
+		if result.err != nil {
+			a.logger.Errorf("error getting secret '%s': %s. Skipping...", result.name, result.err.Error())
+			continue
```

Comment:
is this the desired behavior to skip failed secrets and log only? if so, then pls be sure to add that this is the behavior in the docs pr 🙏